autoemulate/experimental/plotting.py

```diff
@@ -0,0 +1,169 @@
+import matplotlib.pyplot as plt
+import numpy as np
+from IPython.core.getipython import get_ipython
+
+from autoemulate.experimental.types import NumpyLike
+
+
+def display_figure(fig):
+    """
+    Display a matplotlib figure appropriately based on the environment
+    (Jupyter notebook or terminal).
+
+    Args:
+        fig: matplotlib figure object to display
+
+    Returns:
+        fig: the input figure object
+    """
+    # Are we in Jupyter?
+    try:
+        is_jupyter = get_ipython().__class__.__name__ == "ZMQInteractiveShell"
+    except NameError:
+        is_jupyter = False
+
+    if is_jupyter:
+        # we don't show otherwise it will double plot
+        plt.close(fig)
+        return fig
+    # in terminal, show the plot
+    plt.close(fig)
+    plt.show()
+    return fig
+
+
+def plot_Xy(  # noqa: PLR0913
```

Comment:
I think we should stick with lower case naming following the refactor? So e.g. `plot_xy(`